### ISSUE TITLE
TS1259 should not warn/error when used with `import type ...`

### ISSUE THREAD

**Issue by arontsang** [microsoft/TypeScript#48742]
# Bug Report

When using `import type`, the import is fully erased, therefore there is no reason to flag TS1259.

### 🕗 Version & Regression Information

Typescript version 4.6.2

### 💻 Code

```ts
foo.d.ts
import type Foo from './bar'
```

### 🙁 Actual behavior

Foo receives a red underline when viewed in VSCode without a tsconfig.json

### 🙂 Expected behavior

No red underline.

import is fully erased, and in my case, is in a .d.ts file.

**Comment by andrewbranch** [comment]
I’m pretty sure I proposed this in the past and @weswigham had a reason we couldn’t/shouldn’t do it, which I don’t remember enough to articulate here.

**Comment by weswigham** [comment]
Uh, in this case, it's just because as far as we know, a default import is incorrect, given the shape of the library (unless some module loader intervenes - which is what `esModuleInterop` and `allowSyntheticDefaultImports` are for) - there's no declared default member.

**Comment by andrewbranch** [comment]
So what’s the argument for not just effectively assuming `allowSyntheticDefaultImports` in ambient contexts? I could have sworn you gave me one that was more than just correctness; I think you had a scenario where there was a real issue/ambiguity.

**Comment by andrewbranch** [comment]
Something to do with the target module having a member named `default` maybe?

**Comment by RyanCavanaugh** [comment]
Does this have to do with forward-versioning of the target module, e.g. if we start optimistically hoisting-to-default and then the module later adds a default export, it's going to extremely break consumers?

**Comment by weswigham** [comment]
Yep.

**Comment by andrewbranch** [comment]
Sounds like this is working as intended then.
